test/SILOptimizer/existential_transform.swift

```diff
@@ -394,6 +394,76 @@ struct MyStruct : Foo {
   return getName(u) == "MyStruct" ? 0 : 1
 }
 
+protocol RP {
+  var val:Int32 {get set}
+}
+class RC: RP {
+  var val:Int32
+  init(val:Int32) { self.val = val }
+}
+// CHECK-LABEL: sil shared [noinline] @$s21existential_transform4find4base4mult4Obj1Sbs5Int32V_AgA2RP_ptFTf4nne_n : $@convention(thin) <τ_0_0 where τ_0_0 : RP> (Int32, Int32, @in_guaranteed τ_0_0) -> Bool {
+// CHECK: bb0
+// CHECK:   alloc_stack $RP
+// CHECK:   init_existential_addr
+// CHECK:   copy_addr
+// CHECK:   debug_value
+// CHECK:   debug_value
+// CHECK:   debug_value_addr
+// CHECK:   struct_extract
+// CHECK:   struct_extract
+// CHECK:   integer_literal
+// CHECK:   builtin
+// CHECK:   tuple_extract
+// CHECK:   tuple_extract
+// CHECK:   cond_fail
+// CHECK:   open_existential_addr
+// CHECK:   witness_method
+// CHECK:   apply
+// CHECK:   struct_extract
+// CHECK:   builtin
+// CHECK:   cond_br
+// CHECK: bb1:
+// CHECK:   integer_literal
+// CHECK:   struct
+// CHECK:   br
+// CHECK: bb2:
+// CHECK:   open_existential_addr
+// CHECK:   witness_method
+// CHECK:   apply
+// CHECK:   struct_extract
+// CHECK:   builtin
+// CHECK:   cond_br
+// CHECK: bb3
+// CHECK:   dealloc_stack
+// CHECK:   return
+// CHECK: bb4:
+// CHECK:   struct
+// CHECK:   br
+// CHECK: bb5:
+// CHECK:   integer_literal
+// CHECK:   builtin
+// CHECK:   tuple_extract
+// CHECK:   tuple_extract
+// CHECK:   cond_fail
+// CHECK:   struct
+// CHECK:   function_ref @$s21existential_transform4find4base4mult4Obj1Sbs5Int32V_AgA2RP_ptFTf4nne_n : $@convention(thin) <τ_0_0 where τ_0_0 : RP> (Int32, Int32, @in_guaranteed τ_0_0) -> Bool
```

Comment:
@atrick this is where it is checking that the recursive call has been converted into a protocol constrained generic function. Rest of the code is immaterial, to avoid other optimizations to mess it up.

Reply:
Ok. Since you now have a .sil test, you can just refer to that test here, then do you really need to check all this SIL?

I think of the .swift test as: make sure the end-to-end result is still good for this code pattern and the .sil test as make sure this pass itself doesn't break.

Reply:
Thats right.